### PR TITLE
Fix linter

### PR DESCRIPTION
**Pull request by dvonessen**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Done some more linter fixes.

Was the change discussed in an issue?
------------------------------------------------------------

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -4,6 +4,7 @@ doctests = True
 max_line_length = 160
 max_doc_length = 160
 max-complexity = 25
+ignore=E203
```
Comment (ThomasSteinbach) [inline setup.cfg:7]: Where does this lead to problems?
https://www.flake8rules.com/rules/E203.html

Comment (dvonessen) [reply setup.cfg:7]: E203 is a collision between flake8 and black.
https://black.readthedocs.io/en/stable/the_black_code_style.html?highlight=E203#slices

The initial problem occurs [here](https://github.com/dbsystel/gitlab-ci-python-library/blob/fix-linter/gcip/core/cache.py#L181)

Comment (ThomasSteinbach) [reply setup.cfg:7]: Sound fine. I would document this functional requirement here.

```suggestion
# https://black.readthedocs.io/en/stable/the_black_code_style.html?highlight=E203#slices
ignore=E203
```

**Review by ThomasSteinbach — CHANGES_REQUESTED** [review]
When applying the suggestion, you could merge immediately.

File: setup.cfg (hunk 1 above)
Comment (dvonessen) [reply setup.cfg:8]: I added the comment.

**Comment by dvonessen** [comment]
@ThomasSteinbach is this PR approved?

**Pull request merged**